pkg/util/error_list.go

```diff
@@ -52,3 +57,62 @@ func SliceToError(errs []error) error {
 	}
 	return errorList(errs)
 }
+
+// Errors represents an object that may be transformed into a slice of errors by virtue
+// of being an error aggregate (a set of errors grouped together but not part of a parent
+// child relationship).
+type Errors interface {
+	Errors() []error
+}
+
+// ErrMatchFunc is used to match errors.
+type ErrMatchFunc func(error) bool
+
+// matchesError returns true if any ErrMatchFunc returns true
+func matchesError(err error, fns ...ErrMatchFunc) bool {
+	for _, fn := range fns {
+		if fn(err) {
+			return true
+		}
+	}
+	return false
+}
+
+// FilterError returns the error (or, if err implements Errors to indicate it is
+// an error aggregate, the subset of that slice) if all provided match functions
+// return false. This can be used to eliminate known-good errors from a list
+// (such as io.EOF or os.PathNotFound) of errors to determine whether any true
+// error occurred.
+func FilterError(err error, fns ...ErrMatchFunc) error {
+	if list, ok := err.(Errors); ok {
+		return SliceToError(FilterErrors(list.Errors(), fns...))
+	}
+	if matchesError(err, fns...) {
+		return nil
+	}
+	return err
+}
+
+// FilterErrors returns any errors (or nested errors, if the list contains
+// nested Errors) for which all fns return false. If no errors
+// remain a nil list is returned. The resulting silec will have all
+// nested slices flattened as a side effect.
+func FilterErrors(list []error, fns ...ErrMatchFunc) []error {
```

Comment:
Can I beg for FilterOutErrors - FilterErrors sounds like a positive filter, to me.